hs_access_control/tests/test_group_sharing.py

```diff
@@ -131,13 +137,29 @@ def test_02_share_rw(self):
         self.assertTrue(cat.uaccess.can_change_resource(scratching))
         self.assertTrue(cat.uaccess.can_view_resource(scratching))
 
-        # todo: check advanced sharing semantics:
         # should be able to unshare anything one shared.
-
         with self.assertRaises(PermissionDenied) as cm:
             nobody.uaccess.unshare_resource_with_group(scratching, felines)
         self.assertEqual(cm.exception.message, 'Insufficient privilege to unshare resource')
 
+        assertGroupResourceUnshareCoherence(self) 
+
+        self.assertTrue(dog.uaccess.can_unshare_resource_with_group(scratching, felines))
         dog.uaccess.unshare_resource_with_group(scratching, felines)
         self.assertEqual(felines.gaccess.view_resources.count(), 0)
 
+        assertGroupResourceUnshareCoherence(self)
+
+    def test_03_group_share_debugging(self): 
```

Comment:
I should rename it. It was intended to check semantics but it is appropriate to run it always.